core/slice.go

```diff
@@ -586,18 +578,31 @@ func (sl *Slice) updatePhCacheFromDom(pendingHeader types.PendingHeader, termini
 }
 
 // updatePhCache updates cache given a pendingHeaderWithTermini with the terminus used as the key.
-func (sl *Slice) updatePhCache(pendingHeaderWithTermini types.PendingHeader, inSlice bool) {
+func (sl *Slice) updatePhCache(pendingHeaderWithTermini types.PendingHeader, inSlice bool, localHeader *types.Header) {
+	sl.phCacheMu.Lock()
+	defer sl.phCacheMu.Unlock()
+
+	var exists bool
+	if localHeader != nil {
+		termini := sl.hc.GetTerminiByHash(localHeader.ParentHash())
+		pendingHeaderWithTermini, exists = sl.readPhCache(termini[c_terminusIndex])
+		if exists {
+			pendingHeaderWithTermini.Header = sl.combinePendingHeader(localHeader, pendingHeaderWithTermini.Header, common.ZONE_CTX, true)
+		}
+	}
+
 	// Update the pendingHeader Cache
 	oldPh, exist := sl.readPhCache(pendingHeaderWithTermini.Termini[c_terminusIndex])
 	var deepCopyPendingHeaderWithTermini types.PendingHeader
 	newPhEntropy := pendingHeaderWithTermini.Header.CalcPhS()
 	deepCopyPendingHeaderWithTermini = types.PendingHeader{Header: types.CopyHeader(pendingHeaderWithTermini.Header), Termini: pendingHeaderWithTermini.Termini}
 	deepCopyPendingHeaderWithTermini.Header.SetLocation(common.NodeLocation)
 	deepCopyPendingHeaderWithTermini.Header.SetTime(uint64(time.Now().Unix()))
+
 	if exist {
 		// If we are inslice we will only update the cache if the entropy is better
 		if (!inSlice && newPhEntropy.Cmp(oldPh.Header.CalcPhS()) >= 0) ||
-			(inSlice && pendingHeaderWithTermini.Header.ParentEntropy().Cmp(oldPh.Header.ParentEntropy()) > 0) {
+			(inSlice && pendingHeaderWithTermini.Header.ParentEntropy().Cmp(oldPh.Header.ParentEntropy()) >= 0) {
```

Comment:
If parent entropy == new entropy, I don't think we want to update. I think we only want to update if new entropy is a reduction of the parent... right?

Reply:
What is the scenario where a child block has the same entropy as the parent? This is only possible if they didn't do any PoW.